### PR TITLE
Update `PLE2510`, `PLE2512-2515` to check in f-strings

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

This PR updates `PLE2510`, `PLE2512-2515` to check in f-strings.

> ### Reference:
> * `PLE2510`: Invalid unescaped character backspace, use "\b" instead
> * `PLE2512`: Invalid unescaped character SUB, use "\x1A" instead
> * `PLE2513`: Invalid unescaped character ESC, use "\x1B" instead
> * `PLE2514`: Invalid unescaped character NUL, use "\0" instead
> * `PLE2515`: Invalid unescaped character zero-width-space, use "\u200B" instead

## Test Plan

Add test cases for f-strings.

### CODE REVIEW

**Comment by dhruvmanila** [comment]
Current dependencies on/for this PR:
* main
  * **PR #7376** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/7376" target="_blank">Graphite</a> 
    * **PR #7387** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/7387" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/7387?utm_source=stack-comment).

File: crates/ruff/resources/test/fixtures/pylint/invalid_characters.py
Comment (MichaReiser) [inline crates/ruff/resources/test/fixtures/pylint/invalid_characters.py:1]: Uh, why is this a binary file :confused:

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff/resources/test/fixtures/pylint/invalid_characters.py
Comment (dhruvmanila) [reply crates/ruff/resources/test/fixtures/pylint/invalid_characters.py:1]: Weird, locally I don't see it:
```console
$ ls -la invalid_characters*
-rw-r--r--@ 1 dhruv  staff   1.4K Sep 15 12:02 invalid_characters.py
```

I guess it's GitHub getting confused although not sure. This is what I see inside the editor:

Screenshot 2023-09-15 at 12 02 44

Comment (dhruvmanila) [reply crates/ruff/resources/test/fixtures/pylint/invalid_characters.py:1]: And is also visible in the branch: https://github.com/astral-sh/ruff/blob/dhruv/issue-7296/crates/ruff/resources/test/fixtures/pylint/invalid_characters.py

**Pull request merged**